src/System.Management.Automation/utils/ClrFacade.cs

```diff
@@ -60,7 +60,7 @@ internal static IEnumerable<Assembly> GetAssemblies(string namespaceQualifiedTyp
         {
             return PSAssemblyLoadContext.GetAssembly(namespaceQualifiedTypeName) ??
                    AppDomain.CurrentDomain.GetAssemblies().Where(a =>
-                       !TypeDefiner.DynamicClassAssemblyName.Equals(a.GetName().Name, StringComparison.Ordinal));
+                       !a.FullName.StartsWith(TypeDefiner.DynamicClassAssemblyName, StringComparison.Ordinal));
```

Comment:
This will make an edge case stop working -- say the assembly `a` has the name `PowerShell Class Assembly blah, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null`, and it will be filtered out.
Yes, it's an absolute edge case, but we'd better not break it :)

The fix is easy:
```
const string PowerShellClassAssemblyFullNamePrefix = TypeDefiner.DynamicClassAssemblyName + ",";
return ...
           !a.FullName.StartsWith(PowerShellClassAssemblyFullNamePrefix, StringComparison.Ordinal));
```
You can still have comma `,` in your assembly name, but in the `FullName` a backslash will be added before the comma, so we are good.
```
PS:74> $a = [System.Reflection.Assembly]::LoadFrom("$pwd\PowerShell Class Assembly,.dll")
PS:75> $a.FullName
PowerShell Class Assembly\,, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
```

Reply:
I thought it would be harder :-) Many thanks!

Fixed.